feat: adding support for --component flag

skiano opened the pull request:
This is an attempt to resolve https://github.com/egoist/poi/issues/243

I tried to just follow what you outlined in the issue. Let me know what you think 😄

egoist commented on packages/poi/test/config.test.js:
```diff
@@ -197,4 +201,21 @@ describe('get webpack config', () => {
   })
+
+  describe('build component', () => {
+    it('respects component option', async () => {
```
I think we can just require `html-webpack-plugin` here?

skiano replied:
that would probably be clearer, yes

egoist commented on packages/poi/lib/handle-options.js:
```diff
@@ -9,6 +9,14 @@ const { inferHTML, readPkg } = require('./utils')
 
+  if (options.component) {
+    console.log('> Bundling component')
+    options.format = 'cjs'
```
what about supporting `umd` format as well, like:

```bash
poi --component {LibraryName}
```

When `options.component` is a string we build it in UMD format.

skiano replied:
Ok, sounds good

skiano commented:
I added support for `poi --component {LibraryName}` and added an error if you pass `umd` but no module name and an error if you pass anything but `umd` or `cjs`

egoist commented:
The `LibraryName` here actually means `moduleName` that we use in `umd` format 😅 

So `--component foo` should be taken as `--format umd --module-name foo`

skiano commented:
Oh I see! So no name means use `cjs`, but if you put a name, it implies umd. I can make those changes

skiano commented:
@egoist did those changes look how you expect?

egoist commented on packages/poi/package.json:
```diff
@@ -19,6 +19,7 @@
   "bin": "bin/cli.js",
   "scripts": {
     "test": "jest",
+    "test:watch": "jest --watch",
```
`yarn test --watch` or use `npx` will do the trick 😄

egoist commented:
Hmm I'm not sure if we should set `minimize: false` by default for *component*

skiano commented:
@egoist perhaps, but for umd wouldn't you want it `true` by default? I guess i was thinking it is easy enough to set it to `false` yourself.

Also if someone doesn't think about it, isn't it safer for it to be `true`

skiano commented:
@egoist I totally agree. I think I just did that based on the issue conversation. But wouldnt mind for it to be true by default.
